app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -51,47 +51,56 @@ class RoutePresenter(
     private val settingStore: SettingStore = SettingStore.shared,
     private val dataStore: DataStore = DataStore.shared,
     private val accountStore: AccountStore = AccountStore.shared,
-    private val lifecycleStore: LifecycleStore = LifecycleStore.shared
+    private val autoLockStore: AutoLockStore = AutoLockStore.shared
 ) : Presenter() {
     private lateinit var navController: NavController
 
     override fun onViewReady() {
         navController = Navigation.findNavController(activity, R.id.fragment_nav_host)
 
-        // Make the routing contingent on the state of the dataStore.
-        val dataStoreRoutes = dataStore.state
-            .map(this::dataStoreToRouteActions)
-            .filterNotNull()
+        val lockObservable = accountStore.syncCredentials
+            .switchMap {
+                if (it.value != null) {
+                    autoLockStore.lockRequired
+                } else {
+                    Observable.empty<Boolean>()
+                }
+            }
+            .map { if (it) DataStoreAction.Lock else DataStoreAction.Unlock }
 
         // Moves credentials from the AccountStore, into the DataStore.
         accountStore.syncCredentials
             .map(this::accountToDataStoreActions)
-            .filterNotNull()
+            .mergeWith(lockObservable)
             .subscribe(dispatcher::dispatch)
             .addTo(compositeDisposable)
 
+        // Make the routing contingent on the state of the dataStore.
+        val dataStoreRoutes = dataStore.state
+            .map(this::dataStoreToRouteActions)
+
         routeStore.routes
             .mergeWith(dataStoreRoutes)
             .observeOn(mainThread())
             .subscribe(this::route)
             .addTo(compositeDisposable)
     }
 
-    private fun dataStoreToRouteActions(storageState: State): Optional<RouteAction> {
+    private fun dataStoreToRouteActions(storageState: State): RouteAction {
         return when (storageState) {
             is State.Unlocked -> RouteAction.ItemList
             is State.Locked -> RouteAction.LockScreen
             is State.Unprepared -> RouteAction.Welcome
             else -> RouteAction.LockScreen
-        }.asOptional()
```

Comment:
I think we might need to add this back in when @eliserichards starts sending more `State.Errored` actions (not necessarily appropriate to lock when we're offline IMO)